ldap.js

```diff
@@ -83,7 +114,22 @@ exports.getUser = (username, cb) => {
       if (!ret.length) {
         return cb(null, null);
       } else {
-        return cb(null, ret[0]);
+        searchGroups(ret[0].uid, function(err, groups) {
+          if(err) {
+            return cb(null, null);
+          } else {
+            var admin = false;
+            for(var i = 0; i < groups.length; i++) {
+              if(groups[i] === "atlas-admin") {
+                admin = true;
+                break;
+              }
+            }
+            if(admin) ret[0].admin = true;
+            else ret[0].admin = false;
+            return cb(null, ret[0]);
```

Comment:
I don't understand node, express nor its authentication. Are we saving this on the server or can a user manipulate their cookies/requests/session locally and pretend to be an admin?

Reply:
Ah, I'm storing it in the session! What do you think is a good way to avoid it? Perhaps check if a user belongs to `admin-atlas' using ldap each time we need to? 

I tried to find a way to get all users in a group using ldapjs, but couldn't find any. However, there's some way to do it using the ActiveDirectory library. Would you like me to use it and store all admins on the server?

Reply:
Is the session stored server side or client side? Can a user manipulate their own session?

I really don't want to use ActiveDiretory library here, that's recipe for disaster. To get all users in a group, all you have to do is check all the members that belong to a group in our LDAP.

Reply:
Looks like each session has some session ID on the server to prevent forgery. I'm getting my information from [this page](https://stackoverflow.com/questions/5522020/how-do-sessions-work-in-express-js-with-node-js). If that's the case, our code should be secure. :)

Reply:
`express-session` stores session information on the server and only a cookie on the client, so using this with TLS should be secure enough. According to [security best practices for express](https://expressjs.com/en/advanced/best-practice-security.html), we should be setting the cookie name [here](https://github.com/HelioStrike/openmrs-contrib-atlas/blob/a14678f954026797f5bf9a198fb71b29dbe4fa67/app.js#L28-L33) to something other than the default.